tests/ftests/075-pybindings-cgroup_compare_cgroup.py

```diff
@@ -0,0 +1,199 @@
+import ftests
+import sys
+import os
+
+CGNAME = '075cgcompare/compare'
+CONTROLLERS = ['cpu', 'memory', 'io', 'pids']
+
+
+def prereqs(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    if config.args.container:
+        result = consts.TEST_SKIPPED
+        cause = 'This test cannot be run within a container'
+        return result, cause
+
+    if Cgroup.cgroup_mode() != Mode.CGROUP_MODE_UNIFIED:
+        result = consts.TEST_SKIPPED
+        cause = 'This test requires the unified cgroup v2 hierarchy'
+        return result, cause
+
+    return result, cause
+
+
+def setup(config):
+    CgroupCli.create(config, CONTROLLERS, CGNAME)
+
+
+def test(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    #
+    # Test 1 - Compare matching empty cgroup instances
+    #
+    cgempty1 = Cgroup(CGNAME, Version.CGROUP_V2)
+    cgempty2 = Cgroup(CGNAME, Version.CGROUP_V2)
+
+    if cgempty1 != cgempty2:
+        result = consts.TEST_FAILED
+        tmp_cause = 'Empty cgroups do not match'
+        cause = '\n'.join(filter(None, [cause, tmp_cause]))
+
+    #
+    # Test 2 - Compare different empty cgroup instances
+    #
+    cgempty3 = Cgroup('foo', Version.CGROUP_V2)
+    cgempty4 = Cgroup('bar', Version.CGROUP_V2)
+
+    if cgempty3 == cgempty4:
+        result = consts.TEST_FAILED
+        tmp_cause = 'Empty cgroups erroneously match'
+        cause = '\n'.join(filter(None, [cause, tmp_cause]))
+
+    #
+    # Test 3 - Compare different empty cgroup instances
+    #
+    cgempty5 = Cgroup('baz', Version.CGROUP_V1)
+    cgempty6 = Cgroup('baz', Version.CGROUP_V2)
+
+    if cgempty5 == cgempty6:
+        result = consts.TEST_FAILED
+        tmp_cause = 'Empty cgroups erroneously match'
+        cause = '\n'.join(filter(None, [cause, tmp_cause]))
+
+    #
+    # Test 4 - Compare cgroups with only the controllers populated
+    #
+    cgctrl1 = Cgroup(CGNAME, Version.CGROUP_V2)
+    cgctrl1.add_all_controllers()
+
+    cgctrl2 = Cgroup(CGNAME, Version.CGROUP_V2)
+    for controller in CONTROLLERS:
+        cgctrl2.add_controller(controller)
+
+    if cgctrl1 != cgctrl2:
+        result = consts.TEST_FAILED
+        tmp_cause = 'Controller-only cgroups do not match'
+        cause = '\n'.join(filter(None, [cause, tmp_cause]))
+
+    #
+    # Test 5 - Compare populated cgroups
+    #
+    cgget1 = Cgroup(CGNAME, Version.CGROUP_V2)
+    cgget1.add_all_controllers()
+    cgget1.get()
+
+    cgget2 = Cgroup(CGNAME, Version.CGROUP_V2)
+    cgget2.get()
+
+    if cgget1 != cgget2:
+        result = consts.TEST_FAILED
+        tmp_cause = 'cgget1 and cgget2 cgroups do not match'
+        cause = '\n'.join(filter(None, [cause, tmp_cause]))
+
+    #
+    # Test 6 - Compare cgroups with differing uid/gids
+    #
+    cguid1 = Cgroup(CGNAME, Version.CGROUP_V2)
+    cguid1.set_uid_gid(123, 456, 789, 987)
+    cguid1.add_all_controllers()
+
+    cguid2 = Cgroup(CGNAME, Version.CGROUP_V2)
+    cguid2.set_uid_gid(123, 456, 789, 9876)
```

Comment:
Following the convention in the below cases, append `0` to `ctrl_gid` too, instead of `6`. Feel free to ignore this suggestion, it is more of a preference comment.